.github/workflows/main.yml

```diff
@@ -1,18 +1,24 @@
-name: Ruby
+name: Ruby Tests
 
-on: [push,pull_request]
+on: [push, pull_request]
 
 jobs:
-  build:
+  test:
+    name: Ruby ${{ matrix.ruby-version }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['3.0.1', '3.1.1', '3.2.1', '3.3.1', '3.4.1']
+      fail-fast: false
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Ruby
+    - uses: actions/checkout@v4
+    - name: Set up Ruby ${{ matrix.ruby-version }}
       uses: ruby/setup-ruby@v1
       with:
-        ruby-version: 2.7.3
-    - name: Run the default task
-      run: |
-        gem install bundler -v 2.2.3
```

Comment:
`setup-ruby` installs by default the provided bundler version or the one available on the `*.gemspec` file. The ones used to bundle is not compatible with ruby versions below 3.1 so went with the latest compatible version with ruby `3.0.1`.